staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -361,6 +361,9 @@ func finishNothing(context.Context, bool) {}
 
 // Create inserts a new item according to the unique key from the object.
 func (e *Store) Create(ctx context.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
+	// Make a copy so we never mutate the passed-in object (e.g. in tests)
+	obj = obj.DeepCopyObject()
```

Comment:
this commit is ... controversial https://github.com/kubernetes/kubernetes/pull/103545#issuecomment-895457403

Reply:
yes, I will back-populate whatever we decide there :)